src/Cli/dotnet/Installer/Windows/TimestampedFileLogger.cs

```diff
@@ -24,7 +24,7 @@ internal class TimestampedFileLogger : SetupLoggerBase, IDisposable, ISetupLogge
         /// <summary>
         /// Thread safe queue use to store incoming log request messages.
         /// </summary>
-        private ConcurrentQueue<string> _messageQueue = new ConcurrentQueue<string>();
+        private BlockingCollection<string> _messageQueue = new BlockingCollection<string>();
```

Comment:
Since no need to update, maybe it can be `readonly`

Reply:
Done in https://github.com/dotnet/sdk/pull/21085/commits/15cabbb8ac38e066a5dd028ce289098d64e1a141